dapp_runner/descriptor/dapp.py

```diff
@@ -44,21 +47,84 @@ def __ports_factory(value: str) -> PortMapping:  # type: ignore [misc]  # noqa
     ports: List[PortMapping] = field(metadata={"factory": __ports_factory})
 
 
+@dataclass
+class CommandDescriptor:
+    """Exeunit command descriptor."""
+
+    cmd: str = EXEUNIT_CMD_RUN
+    params: Dict[str, Any] = field(default_factory=dict)
+
+
+class _CommandDescriptorList:
+    """Preprocessor for the exescript commands."""
+
+    def _process_command(self, c):
+        if isinstance(c, list):
+            # assuming it's a `run`
+            self.commands.append(CommandDescriptor(params={"args": c}))
+        elif isinstance(c, dict):
+            for cmd, params in c.items():
+                if cmd == EXEUNIT_CMD_RUN and isinstance(params, list):
+                    # support shorthand `run` notation:
+                    # - run:
+                    #    - ["/golem/run/simulate_observations_ctl.py", "--start"]
+                    params = {"args": params}
+                self.commands.append(CommandDescriptor(cmd=cmd, params=params))
+        else:
+            raise DescriptorError(f"Cannot parse the command descriptor `{c}`.")
+
+    def __init__(self):
+        self.commands = list()
+
+    @classmethod
+    def load_commands(cls, value) -> List[CommandDescriptor]:
+        """Load the contents of the commands list."""
+        commands_list = cls()
+
+        if len(value) > 0 and isinstance(value[0], str):
+            # support single line definitions, e.g. `init: ["/docker-entrypoint.sh"]`
+            value = [value]
+
+        for c in value:
+            commands_list._process_command(c)
+
+        return commands_list.commands
+
+
 @dataclass
 class ServiceDescriptor(BaseDescriptor["ServiceDescriptor"]):
     """Yapapi Service descriptor."""
 
     payload: str
-    entrypoint: List[List[str]]
+    init: List[CommandDescriptor] = field(
+        metadata={"load": _CommandDescriptorList.load_commands}, default_factory=list
+    )
+    entrypoint: List[List[str]] = field(default_factory=list)
     network: Optional[str] = None
     ip: List[str] = field(default_factory=list)
     http_proxy: Optional[HttpProxyDescriptor] = None
     depends_on: List[str] = field(default_factory=list)
 
     def __validate_entrypoint(self):
         if self.entrypoint:
+            warnings.warn(
+                f"`{type(self).__name__}.entrypoint` is deprecated. "
+                f"Please use `init` instead.",
+                DeprecationWarning,
+            )
+            if self.init:
+                raise DescriptorError(
+                    "Cannot specify both `init` and `entrypoint`. "
+                    "Please use `init` only."
+                )
             if isinstance(self.entrypoint[0], str):
                 self.entrypoint = [self.entrypoint]  # noqa
+            for c in self.entrypoint:
+                self.init.append(
```

Comment:
I think if we just rename the variable, at this point it would simplify the implementation and I don't think the API inconsistency at this point would affect users

Reply:
as above